### PR TITLE
fix: use base model name for type comparison

### PR DESCRIPTION
**Pull request by larry-x-yu**
**Summary**

1. Use base model name in comparison when getting existing child RecordData

**Testing**
`yarn test` passed locally

### CODE REVIEW

File: addon/record-data.js
Comment (hjdivad) [inline addon/record-data.js:1]: As we discussed OOB, let's instead use `computeBaseModelName` in `_getExistingChildRecordData`.

Although that *should* be redundant it's a helpful safety check.

**Review by hjdivad — CHANGES_REQUESTED** [review]

File: addon/record-data.js
Comment (larry-x-yu) [reply addon/record-data.js:1]: Updated based on our discussion.

File: addon/record-data.js [hunk 1]
```diff
@@ -1055,6 +1055,19 @@ export default class M3RecordData {
...
+     * This is a workaround for now.
+     * As m3 always get the existing child record data from _baseRecordData, thus it
+     * makes sense to use the base model name if it is available; Otherwise just use
+     * the model name - it might be already the base one
+     */
+    let childBaseModelName = (newType && this._schema.computeBaseModelName(newType)) || newType;
+    if (childBaseModelName) {
+      // this is userland API so we have to normalize the name via dasherization
+      newType = dasherize(childBaseModelName);
+    }
```
Comment (hjdivad) [inline addon/record-data.js:1069]: are we sure we only have to do this for `newType` and not the existing type?

Comment (hjdivad) [inline addon/record-data.js:1069]: can we add a test for this?

**Pull request merged**